### PR TITLE
Various improvements to `Markdown` stubs

### PR DESCRIPTION
**Pull request by oprypin**

### CODE REVIEW

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks! A few things I noticed below.

File: stubs/Markdown/markdown/core.pyi [hunk 1]
```diff
@@ -37,27 +37,27 @@ class Markdown:
         self,
         *,
         extensions: Sequence[str | Extension] | None = ...,
-        extension_configs: Mapping[str, Mapping[str, Any]] | None = ...,
+        extension_configs: Mapping[str, dict[str, Any]] | None = ...,
```
[the remark this answers is not in the document]
Comment (oprypin) [reply stubs/Markdown/markdown/core.pyi:40]:
```suggestion
        extension_configs: Mapping[str, Mapping[str, Any]] | None = ...,
```

File: stubs/Markdown/markdown/core.pyi [hunk 2]
```diff
@@ -37,27 +37,27 @@ class Markdown:
...
         self, input: str | _ReadableStream | None = None, output: str | _WritableStream | None = None, encoding: str | None = None
-    ) -> Markdown: ...
+    ) -> Self: ...
 
 def markdown(
     text: str,
     *,
     extensions: Sequence[str | Extension] | None = ...,
-    extension_configs: Mapping[str, Mapping[str, Any]] | None = ...,
+    extension_configs: Mapping[str, dict[str, Any]] | None = ...,
```
[the remark this answers is not in the document]
Comment (oprypin) [reply stubs/Markdown/markdown/core.pyi:60]:
```suggestion
    extension_configs: Mapping[str, Mapping[str, Any]] | None = ...,
```

File: stubs/Markdown/markdown/core.pyi [hunk 3]
```diff
@@ -67,7 +67,7 @@ def markdownFromFile(
     output: str | _WritableStream | None = ...,
     encoding: str | None = ...,
     extensions: Sequence[str | Extension] | None = ...,
-    extension_configs: Mapping[str, Mapping[str, Any]] | None = ...,
+    extension_configs: Mapping[str, dict[str, Any]] | None = ...,
```
[the remark this answers is not in the document]
Comment (oprypin) [reply stubs/Markdown/markdown/core.pyi:70]:
```suggestion
    extension_configs: Mapping[str, Mapping[str, Any]] | None = ...,
```

**Review by srittau — APPROVED** [review]

**Pull request merged**